sdk/typescript/src/signers/signer.ts

```diff
@@ -31,4 +36,61 @@ export abstract class Signer {
    * Returns the signature for the data and the public key of the signer
    */
   abstract signData(data: Base64DataBuffer): Promise<SignaturePubkeyPair>;
+
+  // Returns a new instance of the Signer, connected to provider.
+  // This MAY throw if changing providers is not supported.
+  abstract connect(provider: Provider): Signer;
```

Comment:
i think we should remove this API in favor of having users construct a new instance

Reply:
I thought about this and think this is pretty convenient in the case I mentioned above where the Signer only needs to be connected to a provider when it's necessary. If we do not expose this connect method, the caller needs to load the keypair again(because the keypair is private). Ethers.js has a similar interface: https://github.com/ethers-io/ethers.js/blob/948f77050dae884fe88932fd88af75560aac9d78/packages/abstract-signer/src.ts/index.ts#L81

Reply:
OK, that makes sense - this actually does end up just calling a constructor, so I think it's ok, as long as it isn't changing an existing object